subsys/bluetooth/mesh/transport.c

```diff
@@ -656,13 +671,45 @@ static int sdu_recv(struct bt_mesh_net_rx *rx, u32_t seq, u8_t hdr,
 					  rx->ctx.recv_dst, seq,
 					  BT_MESH_NET_IVI_RX(rx));
 		if (err) {
-			BT_ERR("Unable to decrypt with DevKey");
+			BT_WARN("Unable to decrypt with local DevKey");
+		} else {
+			rx->ctx.app_idx = BT_MESH_KEY_DEV_LOCAL;
+			bt_mesh_model_recv(rx, &sdu);
+			return 0;
+		}
+
+#if defined(CONFIG_BT_MESH_PROVISIONER)
```

Comment:
It should be possible to put this inside a `if (IS_ENABLED(CONFIG_BT_MESH_PROVISIONER)) {` block and thereby avoid the #ifdef.